src/dnf-plugins/product-id/test-productdb.c

```diff
@@ -235,7 +238,7 @@ void testWriteFile(dbFixture *fixture, gconstpointer ignored) {
     gchar *path = g_file_get_path(testJsonFile);
     db->path = path;
 
-    writeProductDb(db, &err);
+    writeProductDb(db);
```

Comment:
Following my suggestion above on `writeProductDb`, this can then:
- `g_assert_true()` on the return value of `writeProductDb`
- `g_assert_no_error()` on `err`, to validate it has no error

Reply:
We can assert here that returned value is 0.